source/includes/connection-snippets/x509-snippets/ruby-connection.rb

```diff
@@ -1,17 +1,11 @@
-# frozen_string_literal: true
-
 # begin x509 connection
 require 'mongo'
+certificate_key_file_path = '/etc/certs/mongodb/client.pem'
+uri = 'mongodb+srv://<cluster-url>/test?authSource=$external&tlsCertificateKeyFile=' + certificate_key_file_path + '&retryWrites=true&w=majority&authMechanism=MONGODB-X509'
```

Comment:
@mongomoe: In https://github.com/mongodb/docs-ecosystem/pull/573#discussion_r349805444 you mentioned that `authSource=$external` was required for "other drivers". Do you know exactly what drivers those are? Our [auth spec's X509 section](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#mongodb-x509) states that the default value is `$external` for the X509 mechanism, so I would expect all drivers to correctly enforce that default if they comply with the spec.

If this is not needed for any driver, then I would propose we remove this.

Separate note regarding `retryWrites=true`. Any driver that supports `tls*` URI options ([DRIVERS-557](https://jira.mongodb.org/browse/DRIVERS-557)) should also have implemented `retryWrites=true` by default ([DRIVERS-531](https://jira.mongodb.org/browse/DRIVERS-531)), so I think that is also redundant. Both of those issues were scheduled for 4.2-compat releases.

Reply:
So for `authSource=$external` I don't currently have a complete list of the drivers that don't work without this (I can eventually put together a sheet of the inconsistencies amongst drivers - currently I just don't have the time for it). But I can definitely confirm that at the very least go doesn't work with it (just using go as an example because that's a language that I'm fairly comfortable with and currently have pulled up). For instance if I use the following url 
`mongodb+srv://balboaclusterm10-5sfvt.mongodb.net/test?tlsCertificateKeyFile=%s&authMechanism=MONGODB-X509` 

I get the error:

# error:
`auth source must be $external `

Reply:
This appears to be a bug in the Go driver then. Since auth source ought not to be required, I would say all drivers where it is in fact not required shouldn't have it in the example code.

Reply:
If `certificate_key_file_path` is given as a variable, it should be escaped with e.g. `URI.encode`.

Reply:
> This appears to be a bug in the Go driver then. Since auth source ought not to be required, I would say all drivers where it is in fact not required shouldn't have it in the example code.

@mongomoe alluded to this in https://github.com/mongodb/docs-ecosystem/pull/584#discussion_r352800377. The immediate goal is to keep the examples consistent for all languages and punt on this until [DOCSP-7932](https://jira.mongodb.org/browse/DOCSP-7932). The fact that this is an outstanding bug in Go will unfortunately affect all language examples for the time being.

> If certificate_key_file_path is given as a variable, it should be escaped with e.g. URI.encode.

@p-mongo: I originally raised this point in https://github.com/mongodb/docs-ecosystem/pull/583#discussion_r352678773 but backed down after realizing that it was only moved to a variable for readability. I agree that it's technically correct to encode any credentials or query string values being inserted into the URI, but it may be out of scope for these simple examples. If the goal is consistency across the various languages, it might be problematic to have to source an external library for languages that don't have encoding in their standard library (e.g. C).